### DISCUSSION TITLE
Does use a GitHub action to update an article on a Jamstack site violate the GitHub Terms of Service?

### DISCUSSION THREAD

**Discussion by nyamadori** [community/community#27082] [Actions]
<p>I create a media site linked to an external CMS. The media site is built with the Jamstack architecture, and GitHub Actions is used to build and deploy the pages of the Jamstack site.</p>
<p>In this architecture, after updating CMS articles, the page needs to be built and deployed to deliver the latest articles. This can be achieved by using the Scheduled Action of GitHub Actions and executing the build workflow periodically (e.g every 0 am).</p>
<p>However, using GitHub Actions only for updating articles without changing the site code (HTML, CSS, JS, etc.) seems to be a serverless computing use beyond the scope of CI/CD. This seems to violate the GitHub Actions Terms of Service (<a href="https://docs.github.com/en/github/site-policy/github-additional-product-terms#4-pages" rel="nofollow noopener">https://docs.github.com/en/github/site-policy/github-additional-product-terms#4-pages</a>).</p>
<p>Is using GitHub Actions for article updates a problem with terms of use?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/nyamadori">@nyamadori</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>No, the terms of use referred is talking about the <a href="https://docs.github.com/en/github/working-with-github-pages" rel="nofollow noopener">Github pages</a>, which is not related to your artical site update.</p>
<p>Github actions mostly act the same as your local machine. Hence, firstly please try to figure out how to use command on local machine to update an article on a Jamstack site, and then you can transfer the code to github workflow.</p>
<p>Thanks</p>

**Comment by nyamadori** [comment]
<p>Thanks for your reply!<br>
I’m sorry, the Terms of Use link I posted was incorrect.</p>
<p>Correctly:<br>
<a href="https://docs.github.com/en/github/site-policy/github-additional-product-terms#a-actions-usage" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/github/site-policy/github-additional-product-terms#a-actions-usage</a></p>
<p>I’m afraid that I’m not violating the following text in the above terms and conditions:</p>
<blockquote>
<p>Additionally, Actions should not be used for:</p>
<ul>
<li>serverless computing;</li>
</ul>
</blockquote>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/nyamadori">@nyamadori</a>,</p>
<p>Thanks for your reply!</p>
<p>Yes, please don’t use github actions for serverless computing according to the product terms.<br>
If you just use it to update articles (without serverlesss computing), please go ahead. :grinning:</p>

**Comment by nyamadori** [comment]
<p>I’ll ask a question to clear up my understanding.</p>
<p>Is “serverless computing” a process that does not reference the Git repository or the code in the repository?</p>

**Comment by tajmone** [comment]
<p>I think that <a class="mention" href="/nyamadori">@nyamadori</a>’s question and concerns are a good point, the Terms of Use document seems to be vague regarding what constitutes “serverless computing” (a term which, in itself, could be subject of multiple interpretations), and (IMO) adding to the document a few practical examples of violating uses (and possibly of legitimate ones) could help clarify the difference.</p>

**Comment by nyamadori** [comment]
<p>After thinking about this post for a few days, I thought it would be better to post a ticket to GitHub Support for the terms of service instead of asking it on GitHub Community.</p>
<p>I’m concerned about creating confusion in discussions on the GitHub Community, for example, with non-employees giving inappropriate answers to the Terms of Service.</p>
<p>So I’m going to post a ticket to GitHub Support with this post.</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="nyamadori" data-post="7" data-topic="126967">
<div class="title">
<div class="quote-controls"></div>
 nyamadori:</div>
<blockquote>
<p>I’m concerned about creating confusion in discussions on the GitHub Community, for example, with non-employees giving inappropriate answers to the Terms of Service.</p>
<p>So I’m going to post a ticket to GitHub Support with this post.</p>
</blockquote>
</aside>
<p>Please, do share a summary of the reply on this post when you get a reply, or even ask explicit permission to paste the email reply here.</p>
<p>I personally think that your question is a good topic for this forum since it concerns the whole community, but I do understand your concerns about the possibility of adding confusion through unofficial replies (although the GH logo on admins’ avatar icons clarifies whose replies are official).</p>

**Comment by theowenyoung** [comment]
<p>Yes, This is my question also.</p>

**Comment by tajmone** [comment, marked as answer]
<aside class="quote no-group quote-modified" data-username="nyamadori" data-post="3" data-topic="126967">
<div class="title">
<div class="quote-controls"></div>
 nyamadori:</div>
<blockquote>
<p>I’m afraid that I’m not violating the following text in the above terms and conditions:</p>
<blockquote>
<p>Additionally, Actions should not be used for:</p>
<ul>
<li>serverless computing;</li>
</ul>
</blockquote>
</blockquote>
</aside>
<p>The <a href="https://docs.github.com/en/github/site-policy/github-terms-for-additional-products-and-features#actions" rel="noopener nofollow ugc">GitHub Terms for Additional Products and Features</a> has been updated/revised since, and it doesn’t seem to contain the above quoted clause anymore — clearer definitions of what may or may not be done using Actions are now provided.</p>
